### PR TITLE
ENH/FIX: The intrusive turbulent class PPE method has been added for both the FOM and the ROM classes.

### PR DESCRIPTION
**Pull request by saddamhijazi**
The intrusive turbulent class PPE method has been added for both the FOM and the ROM classes. The method to compute the turbulent PPE tensors were lacking. At the reduced order level, the assembly of the corresponding tensor has been added.

### CODE REVIEW

File: src/ITHACA_FOMPROBLEMS/UnsteadyNSTurbIntrusive/UnsteadyNSTurbIntrusive.C [hunk 1]
```diff
@@ -1019,3 +1059,72 @@ List< Eigen::MatrixXd > UnsteadyNSTurbIntrusive::bcVelocityMat(label nModes)
...
+    Eigen::Tensor<double, 3> ct1PPETensor;
+    ct1PPETensor.resize(nPModes, nUModes, nUModes);
+
+    for (label i = 0; i < nPModes; i++)
+    {
+        for (label j = 0; j < nUModes; j++)
+        {
+            for (label k = 0; k < nUModes; k++)
+            {
+                // ct1PPETensor(i, j, k) = fvc::domainIntegrate(2 * Pmodes[i] * (fvc::laplacian(
```
Comment (giovastabile) [inline src/ITHACA_FOMPROBLEMS/UnsteadyNSTurbIntrusive/UnsteadyNSTurbIntrusive.C]: please remove comments

File: src/ITHACA_ROMPROBLEMS/ReducedUnsteadyNSTurbIntrusive/ReducedUnsteadyNSTurbIntrusive.C [hunk 2]
```diff
@@ -221,8 +222,11 @@ int newtonUnsteadyNSTurbIntrusivePPE::operator()(const Eigen::VectorXd& x,
                 j) * aTmp;
         bb = aTmp.transpose() * Eigen::SliceFromTensor(problem->bc2Tensor, 0,
                 j) * aTmp;
-        //fvec(k) = m3(j, 0) - gg(0, 0) - m6(j, 0) + bb(0, 0);
-        fvec(k) = m3(j, 0) + gg(0, 0) - m7(j, 0);
+        nn = aTmp.transpose() * Eigen::SliceFromTensor(problem->cTotalPPETensor, 0,
+                j) * aTmp;
+        //fvec(k) = m3(j, 0) + gg(0, 0) - m6(j, 0) + bb(0, 0);
```
Comment (giovastabile) [inline src/ITHACA_ROMPROBLEMS/ReducedUnsteadyNSTurbIntrusive/ReducedUnsteadyNSTurbIntrusive.C]: also here

File: src/ITHACA_FOMPROBLEMS/UnsteadyNSTurbIntrusive/UnsteadyNSTurbIntrusive.C (hunk 1 above)
Comment (saddamhijazi) [reply src/ITHACA_FOMPROBLEMS/UnsteadyNSTurbIntrusive/UnsteadyNSTurbIntrusive.C]: done

File: src/ITHACA_ROMPROBLEMS/ReducedUnsteadyNSTurbIntrusive/ReducedUnsteadyNSTurbIntrusive.C (hunk 2 above)
Comment (saddamhijazi) [reply src/ITHACA_ROMPROBLEMS/ReducedUnsteadyNSTurbIntrusive/ReducedUnsteadyNSTurbIntrusive.C]: done

**Pull request merged**